From 861c777b1811b2e39669a0f240f6aec5f2948915 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Thu, 14 Apr 2022 06:40:02 +0300
Subject: [PATCH] [SW-81745] gaudi3: update erom linker macros

Linker macros for ext rom application is currently hardcoded as per
gaudi2. This patch makes the macros platform specific.

Change-Id: I81ba6782d5b20b8f183d2508712bc52d33539395
---

diff --git a/app/ext_rom/Makefile b/app/ext_rom/Makefile
index 37ca16e..af3ee57 100644
--- a/app/ext_rom/Makefile
+++ b/app/ext_rom/Makefile
@@ -15,6 +15,7 @@
 CFLAGS += -I $(PREBOOT_ROOT_DIR)/include/ -I $(PREBOOT_ROOT_DIR)/ -I $(PREBOOT_ROOT_DIR)/lib/common/ -I $(PREBOOT_ROOT_DIR)/lib/
 CFLAGS += -I $(HL_DEV_ARCH)/ -I $(HL_DEV_ARCH)/../common/ -I $(SPECS_EMBEDDED_DIR)/$(BASE_DEV_NAME)/
 CFLAGS += -include $(SPECS_EMBEDDED_DIR)/$(SPECS_DEV)/$(SPECS_DEV)_reg_map.h
+CFLAGS += -include $(HL_DEV_ARCH)/dev_ext_rom.h
 CFLAGS += -D DEF_HL_FW_COMP_EROM
 
 CFLAGS += -Os -fdata-sections -ffunction-sections
diff --git a/app/ext_rom/common.h b/app/ext_rom/common.h
index 09e8222..92a91fe 100644
--- a/app/ext_rom/common.h
+++ b/app/ext_rom/common.h
@@ -3,14 +3,4 @@
  * Copyright (C) 2021 HabanaLabs Ltd.
  */
 
-#ifndef COMMON_EXT_ROM_H_
-#define COMMON_EXT_ROM_H_
-
-#define EXT_ROM_START		(0x27FE2000)
-#define EXT_ROM_TEXT_LENGTH	(70*1024)
-/* No header bytes. Hence the text base is same as erom start address */
-#define EXT_ROM_PCIE_TEXT_BASE		(EXT_ROM_START)
-#define EXT_ROM_RECOVERY_TEXT_BASE	(EXT_ROM_START + 0x2000)
-#define EXT_ROM_TPM_TEXT_BASE		(EXT_ROM_START)
-
-#endif /* COMMON_EXT_ROM_H_ */
+#include <dev_ext_rom.h>
diff --git a/arch/gaudi2/arc/dev_ext_rom.h b/arch/gaudi2/arc/dev_ext_rom.h
new file mode 100644
index 0000000..0baf793
--- /dev/null
+++ b/arch/gaudi2/arc/dev_ext_rom.h
@@ -0,0 +1,16 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2022 HabanaLabs Ltd.
+ */
+
+#ifndef DEV_EXT_ROM_H_
+#define DEV_EXT_ROM_H_
+
+#define EXT_ROM_START			(0x27FE2000)
+#define EXT_ROM_TEXT_LENGTH		(70 * 1024)
+/* No header bytes. Hence the text base is same as erom start address */
+#define EXT_ROM_PCIE_TEXT_BASE		(EXT_ROM_START)
+#define EXT_ROM_RECOVERY_TEXT_BASE	(EXT_ROM_START + 0x2000)
+#define EXT_ROM_TPM_TEXT_BASE		(EXT_ROM_START)
+
+#endif
diff --git a/arch/gaudi3/arc/dev_ext_rom.h b/arch/gaudi3/arc/dev_ext_rom.h
new file mode 100644
index 0000000..9724d7f
--- /dev/null
+++ b/arch/gaudi3/arc/dev_ext_rom.h
@@ -0,0 +1,16 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2022 HabanaLabs Ltd.
+ */
+
+#ifndef DEV_EXT_ROM_H_
+#define DEV_EXT_ROM_H_
+
+#define EXT_ROM_START			(0x23040000)
+#define EXT_ROM_TEXT_LENGTH		(128*1024)
+/* No header bytes. Hence the text base is same as erom start address */
+#define EXT_ROM_PCIE_TEXT_BASE		(EXT_ROM_START)
+#define EXT_ROM_RECOVERY_TEXT_BASE	(EXT_ROM_START)
+#define EXT_ROM_TPM_TEXT_BASE		(EXT_ROM_START)
+
+#endif
